theano/__init__.py

```diff
@@ -126,6 +126,12 @@ def test():
         raise ImportError("The nose module is not installed."
                           " It is needed for Theano tests.")
 
+if config.device.startswith('cpu'):
+    try:
+        import theano.sandbox.mkl
+    except Exception as e:
```

Comment:
Sure, we can handle with the known exceptions here, while leave others as a warning instead of a simple "pass" to be more meaningful.